sklearn/linear_model/randomized_l1.py

```diff
@@ -354,6 +354,10 @@ def _randomized_logistic(X, y, weights, mask, C=1., verbose=False,
         X *= (1 - weights)
 
     C = np.atleast_1d(np.asarray(C, dtype=np.float64))
+    if C.ndim > 1:
+        raise ValueError("C should be 1-dimensional array, "
+            "but got %d-dimensional array instead." % C.ndim)
```

Comment:
I forgot: it is always good to show the parameter that you got because it gives an additional hint to the user what he needs to change.

So something like this (add the "a" that were missed from my previous comment):
```py
raise ValueError("C should be a 1-dimensional array-like, "
    "but got a {}-dimensional array-like instead: {}".format(C.ndim, C))
```

It would be great to add a test for this error message by the way ...